packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -305,6 +305,11 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
         if (IBaseTerminalServer.validateId(terminalId)) {
             return terminalId;
         }
+        if (this.options.id) {
```

Comment:
@lmcbout why do we dispose a terminal that have `ids`?

There are likely to be extenders and/or plugins which use our API to create terminals and provide their own `ids`.
With this change those terminals will now be `disposed` which is a breaking behavior.